src/cluster/cluster.cc

```diff
@@ -849,6 +849,21 @@ Status Cluster::CanExecByMySelf(const redis::CommandAttributes *attributes, cons
     return {Status::RedisClusterDown, "Hash slot not served"};
   }
 
+  bool cross_slot_ok = false;
+  if (script_run_ctx) {
+    if (script_run_ctx->current_slot != -1 && script_run_ctx->current_slot != slot) {
+      if (getNodeIDBySlot(script_run_ctx->current_slot) != getNodeIDBySlot(slot)) {
+        return {Status::RedisMoved, fmt::format("{} {}:{}", slot, slots_nodes_[slot]->host, slots_nodes_[slot]->port)};
+      }
```

Comment:
Seems here we'll ignore all other checks below?

Reply:
`RedisMoved` is prohibited for scripts. Non-script calls to `CanExecByMySelf` will not go to this branch.

`RedisGenericCommand`
```cpp
auto s = srv->cluster->CanExecByMySelf(attributes, args, conn, script_run_ctx);
if (!s.IsOK()) {
    if (s.Is<Status::RedisMoved>()) {
        PushError(lua, "Script attempted to access a non local key in a cluster node script");
    } else {
        PushError(lua, redis::StatusToRedisErrorMsg(s).c_str());
    }
    return raise_error ? RaiseError(lua) : 1;
}
```

Reply:
> allow-cross-slot-keys 
> Different slots on different nodes are still prohibited from crossing Slots. However, slots on the same node are allowed to cross slots when this flag is enabled (generally not recommended).

Reply:
Ahh I see.